### PR TITLE
Correctly Handle EVENT_NONE in UnixNetVConnection::acceptEvent

### PR DESCRIPTION
**Pull request by oknet**
The `data == nullptr` is always with `event == EVENT_NONE`.

```
  /** 
    Receives the event code and data for an Event.

    This function receives the event code and data for an event and
    forwards them to the current continuation handler. The processor
    calling back the continuation is responsible for acquiring its
    lock.

    @param event Event code to be passed at callback (Processor specific).
    @param data General purpose data related to the event code (Processor specific).
    @return State machine and processor specific return code.

  */
  int 
  handleEvent(int event = CONTINUATION_EVENT_NONE, void *data = 0)
  {
    return (this->*handler)(event, data);
  }
```

### CODE REVIEW

**Comment by oknet** [comment]
@scw00 
The EVENT_NONE means the callback is not triggered by EventSystem directly.
The data is optional if event is EVENT_NONE.
In the UnixNetVC::acceptEvent, ATS should ignore the data if event is EVENT_NONE.
Because the acceptEvent is designed to put a new NetVC into NetHandler that running in current EThread.

File: iocore/net/UnixNetVConnection.cc [hunk 1]
```diff
@@ -1119,19 +1119,21 @@ UnixNetVConnection::startEvent(int /* event ATS_UNUSED */, Event *e)
...
 
-  MUTEX_TRY_LOCK(lock, get_NetHandler(thread)->mutex, e->ethread);
+  MUTEX_TRY_LOCK(lock, get_NetHandler(t)->mutex, t);
   if (!lock.is_locked()) {
     if (event == EVENT_NONE) {
-      thread->schedule_in(this, HRTIME_MSECONDS(net_retry_delay));
+      t->schedule_in(this, HRTIME_MSECONDS(net_retry_delay));
       return EVENT_DONE;
     } else {
       e->schedule_in(HRTIME_MSECONDS(net_retry_delay));
```
Comment (scw00) [inline iocore/net/UnixNetVConnection.cc]: what do we do if e is nullptr?

**Comment by zwoop** [comment]
Do we need this for 7.1.1 ?

**Comment by zwoop** [comment]
I'm going to cherry-pick this, together with 9ded874c82de732d2c2d42c2b633d7e94adcb2e7

**Pull request merged**